From 38aed9dd19ec8307fe70b3a50097777957e90c70 Mon Sep 17 00:00:00 2001
From: Charlie Sharpsteen <source@sharpsteen.net>
Date: Sat, 11 May 2019 13:13:45 -0700
Subject: [PATCH 07/12] Update libwhereami to build on debian armhf

This commit updates the libwhereami component to build on the debian-9-armhf
and debian-10-armhf platforms using Debian's cmake and to link against
the platform's Boost libraries.
---
 configs/components/libwhereami.rb | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/configs/components/libwhereami.rb b/configs/components/libwhereami.rb
index e997b0f3..3855fea5 100644
--- a/configs/components/libwhereami.rb
+++ b/configs/components/libwhereami.rb
@@ -13,6 +13,23 @@ component "libwhereami" do |pkg, settings, platform|
     cmake = "/usr/local/bin/cmake"
     special_flags = "-DCMAKE_CXX_FLAGS='#{settings[:cflags]}'"
     boost_static_flag = "-DBOOST_STATIC=OFF"
+  elsif platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/
+    # Use system cmake and libraries
+    cmake = '/usr/bin/cmake'
+
+    # Env variables used by CMakeCross.txt to select the desired
+    # crossbuild-essential-<arch> toolchain.
+    pkg.environment('DEB_HOST_ARCH', platform.architecture)
+    pkg.environment('DEB_HOST_GNU_TYPE', platform.platform_triple)
+    toolchain = '-DCMAKE_TOOLCHAIN_FILE=/etc/dpkg-cross/cmake/CMakeCross.txt'
+
+    boost_static_flag = "-DBOOST_STATIC=OFF"
+    special_flags = ["-DBOOST_INCLUDEDIR=/usr/include",
+                     "-DBOOST_LIBRARYDIR=/usr/lib/#{platform.platform_triple}",
+                     "-DCMAKE_FIND_ROOT_PATH='/opt/puppetlabs/puppet;/usr'"]
+
+    special_flags += ["-DENABLE_CXX_WERROR=OFF"] if platform.name =~ /debian-10/
+    special_flags = special_flags.join(' ')
   elsif platform.is_cross_compiled_linux?
     toolchain = "-DCMAKE_TOOLCHAIN_FILE=/opt/pl-build-tools/#{settings[:platform_triple]}/pl-build-toolchain.cmake"
     cmake = "/opt/pl-build-tools/bin/cmake"
-- 
2.20.1 (Apple Git-117)

